### PR TITLE
Add trace origin

### LINKED ISSUE #668: Send trace origin
Trace origin indicates what created a trace or a span. Not all traces and spans contain enough information to tell whether the user or what precisely in the SDK created it. Origin solves this problem. Origin can be sent with the [trace context](https://develop.sentry.dev/sdk/event-payloads/contexts/#trace-context) or [spans](https://develop.sentry.dev/sdk/event-payloads/span/). The SDKs must set this value automatically. Users are not expected to set it manually. For more information, visit the [develop docs](https://develop.sentry.dev/sdk/performance/trace-origin/).

As the origin for trace context and spans is optional, every SDK can decide whether to start sending the origin for everything or just selected transactions and spans. This issue is important for internal data analysis to understand the quality of our traces.

### PR DESCRIPTION
**Pull request by ribice**
Resolves #668

### CODE REVIEW

**Comment by cleptric** [comment]
LGTM, anything else you want to add here?

**Comment by ribice** [comment]
> LGTM, anything else you want to add here?

No, just haven't had the chance to test it e2e.

**Comment by ribice** [comment]
CleanShot 2024-07-03 at 12 30 10

These are visible in spans. Ready for review/merge now.

File: tracing.go [hunk 1]
```diff
@@ -18,6 +18,20 @@ const (
 	SentryBaggageHeader = "baggage"
 )
 
+// SpanOrigin indicates what created a trace or a span. See: https://develop.sentry.dev/sdk/performance/trace-origin/
+type SpanOrigin string
+
+const (
+	SpanOriginManual   = "manual.http"
```
Comment (cleptric) [inline tracing.go:25]: We don't really know if it's an http related span so we can just keep `manual`.

**Review by cleptric — APPROVED** [review]

**Pull request merged**